### ISSUE TITLE
Hosting - Experimental Web Framework Support - Next.js - Emulate Works but Deploy Fails

### ISSUE THREAD

**Issue by jalexw** [firebase/firebase-tools#5250]
### [REQUIRED] Environment info

**firebase-tools:** 11.16.1

**Platform:** macOS 12.4 M1 Pro

**npm:** 9.1.1

**next:** 13.0.3

**react/react-dom:** 18.2.0

**typescript:** 4.9.3

### [REQUIRED] Test case

```sh
npx create-next-app example-project-jalexw
cd example-project-jalexw
firebase init
npm install
firebase deploy --only hosting
```

### [REQUIRED] Steps to reproduce

Create a blank Next.js app
```sh
npx create-next-app example-project
cd example-project
npm install next@latest react@latest react-dom@latest
# Set up Hosting. Use an existing project.
firebase init
# Detected an existing Next.js codebase in the current directory, should we use this? Yes
```

Works 👍 
```sh
firebase emulators:start --only hosting
```

Doesn't work 👎 
```sh
firebase deploy --only hosting
```

### [REQUIRED] Expected behavior

The default Next.js app is emulatable locally and deployable. i.e. https://\<example-project-jalexw\>.web.app:
<img width="1512" alt="Screen Shot 2022-11-16 at 3 32 14 PM" src="https://user-images.githubusercontent.com/51729845/202276495-341399a8-1618-4e1b-aa76-edf69513e703.png">

### [REQUIRED] Actual behavior

Emulating locally works:
```sh
firebase emulators:start --only hosting
```

App appears to build successfully:
```
i  functions: packaged /Users/jalexw/<example-project-jalexw>/.firebase/<example-project-jalexw>/functions (14.46 MB) for uploading
```

But then it fails to deploy after packaging and uploading:
```
Cloud Run service projects/<example-project-jalexw>/locations/us-central1/services/ssr<example-project-jalexw> for the function was not found. The function will not work correctly. Please redeploy
```

```
[debug] [2022-11-16T19:38:13.816Z] TypeError: Cannot read properties of undefined (reading 'service')
    at /Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/functions/backend.js:198:113
    at Array.map (<anonymous>)
    at loadExistingBackend (/Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/functions/backend.js:198:69)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.existingBackend (/Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/functions/backend.js:174:9)
    at async prepare (/Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/functions/prepare.js:151:115)
    at async chain (/Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/index.js:35:9)
    at async deploy (/Users/jalexw/.npm-global/lib/node_modules/firebase-tools/lib/deploy/index.js:79:5)
[error] 
[error] Error: An unexpected error has occurred.
```

**Comment by Hacksore** [comment]
I had this same problem and I deleted the cloud function and deployed again and it fixed it for me, YMMV.

**Comment by jalexw** [comment]
Thanks! That fixed it for me.

The problem was that my first build of the SSR function failed (I hadn't set up credentials for a dependency from a private repository) and `firebase deploy` does not overwrite the failed build. 

I tried again in a blank new project and got another build error from the service account not having the "roles/artifactregistry.reader" permission. I finally got it working after granting that permission, deleting the Cloud Function once again, and redeploying...

I think `firebase-tools` should check if a deployment already failed before trying to deploy the SSR function... The CLI user only sees: `[error] Error: An unexpected error has occurred.` and the details in the logs indicate almost the opposite of what the problem was: `Cloud Run service projects/<example-project-jalexw>/locations/us-central1/services/ssr<example-project-jalexw> for the function was not found.` (in reality the problem was that the service WAS already found?)

I think one of the following should be added:
- A warning message when a build has already failed to tell the user they might need to delete the function before redeploying
- Deploy over the failed build / auto-delete it
